### DISCUSSION TITLE
Extract patent numbers from case law, filings, etc.

### DISCUSSION THREAD

**Discussion by mlissner** [freelawproject/foresight#43] [Ideas]
This is a request I've gotten a number of times over the years, but I've never really looked into how feasible it is. I suppose a regex would get you close, but I imagine it would have a lot of false positives (maybe AI could help eliminate these?). 

Lots of IP lawyers around here. Maybe somebody has thought about this. 

---

Once we've got the numbers extracted:
 - Let people follow them
 - Put them in an API
 - Make them searchable

What else?

**Comment by anseljh** [comment]
Patent numbers are just 7- or 8-digit numbers. There's often context around the number that can help figure out that the number is a patent number, like "U.S. Pat. 1,234,567," but we'd undercount if we relied on those exclusively. It's tricky!

Being able to connect a patent to litigations is quite valuable. Patent defendants always want to know if the patent-in-suit has been previously litigated, and dig into those cases.

Trademark registrations have similar properties. You might see "reg. no. 2,345,678" or you might not.

**Reply by mlissner** [reply]
Isn't there a thing with referring to the patent by the last three digits that we could use? So if `1,234,567` shows up in a doc, we ignore it, but if `'567` shows up later, we know it's a patent number?

**Comment by mlissner** [comment]
Just got another request for this.
